src/aiperf/accuracy/benchmarks/hellaswag.py

```
@@ -1,31 +1,317 @@
         )
+
+    def _build_problems(
+        self, ds: DatasetDict, tasks: list[Any], n_shots: int
+    ) -> list[BenchmarkProblem]:
+        train_set = ds["train"]
+        shots_set = _build_unique_activity_label_shots_set(train_set)
+        val_set = ds["validation"]
+        problems: list[BenchmarkProblem] = []
+        choices = ["A", "B", "C", "D"]
+        # Pre-bucket validation rows by activity_label so the per-task
+        # loop is O(val_rows + tasks) instead of O(tasks × val_rows).
```

Ruff (0.15.13)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)